test/common/http/http1/codec_impl_test.cc

```diff
@@ -628,13 +628,22 @@ TEST_F(Http1ServerConnectionImplTest, IdentityAndChunkedBody) {
   MockRequestDecoder decoder;
   EXPECT_CALL(callbacks_, newStream(_, _)).WillOnce(ReturnRef(decoder));
 
+  TestRequestHeaderMapImpl expected_headers{
+      {":path", "/"},
+      {":method", "POST"},
+      {"transfer-encoding", "identity,chunked"},
```

Comment:
please add an integration test for your use case chunked,chunked because I'm surprised http_parser handles correctly.